src/control/server/server.go

```diff
@@ -436,6 +436,12 @@ func Start(log *logging.LeveledLogger, cfg *config.Server) error {
 	ctx, shutdown := context.WithCancel(context.Background())
 	defer shutdown()
 
+	unloadFabricLibs, err := hwprov.LoadRuntimeFabricLibs(log)
```

Comment:
I'm not sure how I feel about this... I get that you have to load them somewhere early on and hold them open until there's no more need for them. I think the problem I have is that feels a bit like it's exposing too much of the implementation.

What about a `hwprov.Init()` that returns a closure to cleanup? Then you can do other stuff in there later if you need to load other libraries. It seems unlikely to me that the callers at this level will really want/need control over which groups of libraries are dynamically loaded.

Reply:
Good idea. I can rename.